src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java

```diff
@@ -75,12 +88,41 @@ public void process(Exchange exchange) throws Exception {
         }
 
         if (validatedBuildingLocations.size() != jsrValidatedBuildingLocations) {
-            setFileStatus(exchange, applicationContext);
+            setFileStatus(exchange, applicationContext,PARTIAL_SUCCESS);
+        }
+
+        if (buildingLocations.size() > 0) {
+            processExceptionRecords(exchange, buildingLocations);
         }
 
         exchange.getMessage().setBody(validatedBuildingLocations);
+
+    }
+
+
+    private void processExceptionRecords(Exchange exchange,
+                                         List<BuildingLocation> buildingLocationsList) {
+        List<Pair<String, Long>> zeroByteCharacterRecords = new ArrayList<>();
+        buildingLocationsList.forEach(buildingLoc -> dataQualityCheckConfiguration.zeroByteCharacters
```

Comment:
public field dereference, please consider defining method for zeroByteCharacters and accessing via method

Reply:
defined zerobyteCharacterCheck moved zerobyte check to method and added null check

Reply:
zeroBytesCharacters is still being accessed outside of object directly. please introduce 'getter' method and access via method

Reply:
added getters and setters